src/safeposix/syscalls/sys_calls.rs

```diff
@@ -409,16 +409,49 @@ impl Cage {
         0
     }
 
+    /// ### Description
+    ///
+    /// The exec system call replaces the current Cage object image with a new
+    /// Cage object exec is called immediately after a process(Cage) forks.
+    /// For our purposes this translates to removing the parent cage object
+    /// from the cagetable and adding the child object to it. We also close
+    /// any file descriptors that the parent object holds if the `O_CLOEX`
+    /// flag has been set on that file descriptor  
+    ///
+    /// ### Arguments
+    ///
+    /// `child_cageid`: uid of the new child Cage object
+    ///
+    /// ### Returns
+    ///
+    /// Returns 0 upon successfully update the current running image
+    ///
+    /// ### Errors
+    ///
+    /// This syscall doesn't directly have any cases where it returns an error
+    ///
+    /// ### Panics
+    ///
+    /// This function doesn't directly panic - but the unmap memory mappings
+    /// function panics if it cannot create an shm entry
+    ///
+    /// For more information please refer to - [https://man7.org/linux/man-pages/man3/exec.3.html]
     pub fn exec_syscall(&self, child_cageid: u64) -> i32 {
+        // We remove the current running process from the cagetable
         interface::cagetable_remove(self.cageid);
-
+        // Function call to unmap shared memory mappings of the current process
         self.unmap_shm_mappings();
 
+        // Initialize an empty vector to hold file descriptors
         let mut cloexecvec = vec![];
         for fd in 0..MAXFD {
+            // Get mutex of the file descriptor
             let checkedfd = self.get_filedescriptor(fd).unwrap();
```

Comment:
can panic here, should this be documented

Reply:
Yes - thanks for pointing that out!

Reply:
Added!